lib/storage.js

```diff
@@ -31,12 +31,15 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
  *
  * @param {object} options - Options to use.
  * @param {string} options.client_id - The ID of the client.
+ * @param {string} options.token_endpoint - The URL of the token endpoint.
  *
  * @returns {string} The access token if there is one, null if not.
  */
-export async function get({client_id}) {
+export async function get({client_id, token_endpoint}) {
   const collection = database.collections['oauth2-client-accessToken'];
-  const query = {'token.client_id': client_id};
+  const query = {
+    'token.client_id': client_id, 'token.token_endpoint': token_endpoint
```

Comment:
It doesn't look like there's an index for this query, so one needs to be added in this PR.